daemon/daemon.go

```diff
@@ -203,16 +205,30 @@ func (d *Daemon) Run() error {
 	// set image proxy
 	ctrd.SetImageProxy(d.config.ImageProxy)
 
+	httpReadyCh := make(chan bool)
+	criReadyCh := make(chan bool)
+
 	httpServerCloseCh := make(chan struct{})
 	go func() {
-		if err := d.server.Start(); err != nil {
+		if err := d.server.Start(httpReadyCh); err != nil {
 			logrus.Errorf("failed to start http server: %v", err)
 		}
 		close(httpServerCloseCh)
 	}()
 
 	criStopCh := make(chan error)
-	go criservice.RunCriService(d.config, d.containerMgr, d.imageMgr, criStopCh)
+	go criservice.RunCriService(d.config, d.containerMgr, d.imageMgr, criStopCh, criReadyCh)
+
+	httpReady := <-httpReadyCh
+	criReady := <-criReadyCh
+
+	if httpReady && criReady {
```

Comment:
cri manager may don't start if `enable-cri=false` is specified, is it ok in this case?

Reply:
it's ok. If **enable-cri=false** is specified, we can read **true** from thehttpReadyCh